sunpy/net/dataretriever/tests/test_noaa.py

```diff
@@ -1,16 +1,61 @@
 import pytest
 
+from unittest import mock
+
 from sunpy.time import parse_time
 from sunpy.time.timerange import TimeRange
 from sunpy.net.vso.attrs import Time, Instrument
-from sunpy.net.dataretriever.client import QueryResponse
+from sunpy.net.dataretriever.client import QueryResponse, QueryResponseBlock
 import sunpy.net.dataretriever.sources.noaa as noaa
 from sunpy.net.fido_factory import UnifiedResponse
 from sunpy.net import Fido
 from sunpy.net import attrs as a
 LCClient = noaa.NOAAIndicesClient()
 
 
+def create_mock_unified_object(start_date, end_date):
+    '''
+    Let us create a mock QueryResponse object
+    using that, we will construct UnifiedResponse
+    We will prefill some downloaded data from
+    running noaa.NOAAIndicesClient().fetch(Time('2012/8/9', '2012/8/10'),
+                                                    Instrument('noaa-indices'))
+    '''
+    # Create a mock QueryResponse object
+    map_ = {
+        'Time_start': parse_time(start_date),
+        'Time_end':  parse_time(end_date),
+        'source': 'sdic',
+        'instrument': 'noaa-indices',
+        'physobs': 'sunspot number',
+        'provider': 'swpc'
+
+    }
+
+    resp = QueryResponse.create(map_, LCClient._get_default_uri())
+    # Attach the client with the QueryResponse
+    resp.client = 'noaa-indices'
+
+    # Create a UnifiedResponse object
+    uresp = UnifiedResponse(resp)
+    return (uresp)
+
+
+@pytest.mark.remote_data
+def test_fetch_working():
+    '''
+    Tests if the online server for noaa
+    is working.
+    Uses the url : ftp://ftp.swpc.noaa.gov/pub/weekly/RecentIndices.txt
+    '''
+    qr1 = LCClient.search(Time('2012/10/4', '2012/10/6'),
+                          Instrument('noaa-indices'))
+    res = LCClient.fetch(qr1)
+    download_list = res.wait(progress=False)
+    assert len(download_list) == len(qr1)
+    assert download_list[0].split('/')[-1] == 'RecentIndices.txt'
+
```

Comment:
So this is just the old test? Not a test for the mock object?

Reply:
This is the test for checking the old files are same, or have changed.

Reply:
But not tests the mock object against the return?

Reply:
So, I should test the starting date, and ending date?
Because, the final result of `fetch` is a list of filenames?
And with the result of `search`, I can assert the Starting and ending date, or do you have something else in mind?

Reply:
I mean testing the mock result against the real result.